drgn/helpers/linux/mm.py

```diff
@@ -637,12 +637,20 @@ def compound_head(page: Object) -> Object:
     :return: ``struct page *``
     """
     page = page.read_()
-    head = page.compound_head.read_()
+    try:
+        head = page.compound_head.read_()
+    except AttributeError:
+        # RHEL 7
+        head = page.first_page.address_
     if head & 1:
         return cast(page.type_, head - 1)
     # Handle fake head pages (see _page_is_fake_head()).
     if page.flags & (1 << page.prog_["PG_head"]):
-        head = page[1].compound_head.read_()
+        try:
+            head = page[1].compound_head.read_()
+        except AttributeError:
+            # RHEL 7
+            head = page[1].first_page.address_
```

Comment:
My suggestion above removes the need for this change, since we'll never get to this case if `compound_head` didn't exist in the first place.